### PR TITLE
Reworked attachments, nr3 - #2125  #2106 #2132

### PR DESCRIPTION
**Pull request by harshad1**
This is a second bugfix for the attachment PR 

PRs: #2125 #2106 #2132

### CODE REVIEW

**Comment by harshad1** [comment]
@gsantner This is ready to merge IMO

File: app/src/main/java/net/gsantner/opoc/util/GsContextUtils.java [hunk 1]
```diff
@@ -1932,18 +1906,18 @@ public void onReceive(Context context, Intent intent) {
...
      * @param file File that should be edited
      */
-    public void requestPictureEdit(final Context context, final File file) {
-        final Uri uri = getUriByFileProviderAuthority(context, file);
+    public void requestFileEdit(final Context context, final File file) {
+        final Uri uri = FileProvider.getUriForFile(context, getFileProvider(context), file);
         final Intent intent = new Intent(Intent.ACTION_EDIT);
-        intent.setDataAndType(uri, "image/*");
-        intent.addFlags(Intent.FLAG_GRANT_WRITE_URI_PERMISSION | Intent.FLAG_GRANT_READ_URI_PERMISSION);
+        intent.setData(uri);  // Mime type inferred automatically
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/opoc/util/GsContextUtils.java:1916]: Sure that this always works as expected? While the original intent here was to edit pictures and thus image/ was specified, I think it makes sense to keep it this way to explicitly give additional info.

My idea is to use the Fileutils mimetype method here and keep using setDataAndType

File: app/src/main/java/net/gsantner/opoc/util/GsContextUtils.java [hunk 2]
```diff
@@ -1932,18 +1906,18 @@ public void onReceive(Context context, Intent intent) {
...
+        intent.addFlags(Intent.FLAG_GRANT_WRITE_URI_PERMISSION);
+        intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
         intent.putExtra(MediaStore.EXTRA_OUTPUT, uri);
-        intent.putExtra(EXTRA_FILEPATH, file.getAbsolutePath());
```
Comment (gsantner) [inline app/src/main/java/net/gsantner/opoc/util/GsContextUtils.java:1945]: Should be fine to keep sending this bit of info? For some software (like Markor 😄 ) it makes it easier for them to work if they have the full real path.

**Review by gsantner — CHANGES_REQUESTED** [review]

File: app/src/main/java/net/gsantner/opoc/util/GsContextUtils.java (hunk 2 above)
Comment (harshad1) [reply app/src/main/java/net/gsantner/opoc/util/GsContextUtils.java:1945]: Added it back in

File: app/src/main/java/net/gsantner/opoc/util/GsContextUtils.java (hunk 1 above)
Comment (harshad1) [reply app/src/main/java/net/gsantner/opoc/util/GsContextUtils.java]: It seems to work both ways. Using your approach now.

**Review by gsantner — APPROVED** [review]

**Pull request merged**